vision/detect/generated/sample-template.go

```diff
@@ -306,3 +306,60 @@ func detectLogos(w io.Writer, file string) error {
 
 	return nil
 }
+
+// [START vision_detect_async_document{REGION_TAG_PARAMETER}]
+
+// TODO: write docs
+// detectAsyncDocument
+func detectAsyncDocument(w io.Writer, gcs_source_uri, gcs_destination_uri string) error {
+	ctx := context.Background()
+
+	client, err := vision.NewImageAnnotatorClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	feature := &visionpb.Feature{
+		Type: visionpb.Feature_DOCUMENT_TEXT_DETECTION,
+	}
+
+	inputConfig := &visionpb.InputConfig{
+		GcsSource: &visionpb.GcsSource{Uri: gcs_source_uri},
+		// Supported MimeTypes are: "application/pdf" and "image/tiff".
+		MimeType: "application/pdf",
+	}
+
+	outputConfig := &visionpb.OutputConfig{
+		GcsDestination: &visionpb.GcsDestination{Uri: gcs_destination_uri},
+		// How many pages should be grouped into each json output file.
+		BatchSize: 2,
+	}
+
+	asyncRequest := &visionpb.AsyncAnnotateFileRequest{
+		Features: []*visionpb.Feature{feature},
+		InputConfig: inputConfig,
+		OutputConfig: outputConfig,
+	}
+
+	request := &visionpb.AsyncBatchAnnotateFilesRequest{
```

Comment:
I think I'd prefer to have everything in one variable assignment, rather than separate variables for each field.

```go
request := &visionpb.AsyncBatchAnnotateFilesRequest{
	Requests: []*visionpb.AsyncAnnotateFileRequest{
		{
			Features: []*visionpb.Feature{
				{
					Type: visionpb.Feature_DOCUMENT_TEXT_DETECTION,
				},
			},
			InputConfig: ...,
			OutputConfig: ...,
		},
	},
}
```

Reply:
Got it, changed